src/api/dataSet.js

```diff
@@ -9,13 +9,25 @@ export async function getDataSetDetails(dataSetId) {
 
   return {
     ...response,
-    samples: formatSamples(response.data, response.samples),
     experiments: formatExperiments(response.experiments)
   };
 }
 
-export async function updateDataSet(dataSetId, dataSet) {
-  return await Ajax.put(`/dataset/${dataSetId}/`, { data: dataSet });
+export async function updateDataSet(dataSetId, dataSet, details = false) {
+  // Q: is it wrong to include a query parameter in a PUT request?
+  // the `details` parameter is only used to "modify" the reponse from the API, it doesn't
+  // cause any side effect or different behavior
```

Comment:
This is a great question, the answer is no, it's not technically wrong, although it looks a bit confusing because normally the query would be what's set by the PUT. You should change this comment to explain this incongruous behavior. It isn't RESTful though and we can change the API behavior if necessary to clarify this.